mmdet/models/detectors/dab_detr.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Dict, Tuple
+
+import torch
+from mmengine.model import uniform_init
+from torch import Tensor, nn
+
+from mmdet.registry import MODELS
+from ..layers import SinePositionalEncodingHW
+from ..layers.transformer import (DabDetrTransformerDecoder,
+                                  DabDetrTransformerEncoder, inverse_sigmoid)
+from .detr import DETR
+
+
+@MODELS.register_module()
+class DABDETR(DETR):
+    r"""Implementation of `DAB-DETR:
+    Dynamic Anchor Boxes are Better Queries for DETR.
+
+    <https://arxiv.org/abs/2201.12329>`_.
+
+    Code is modified from the `official github repo
+    <https://github.com/IDEA-Research/DAB-DETR>`_.
+
+    Args:
+        iter_update (bool): Whether to update references layer-by-layer.
+            Defaults to True.
+        random_refpoints_xy (bool): Whether to randomly initialize query
+            embeddings and not update them during training.
+            Defaults to False.
+        num_patterns (int): Inspired by Anchor-DETR. Defaults to 0.
+    """
+
+    def __init__(self,
+                 *args,
+                 iter_update: bool = True,
+                 random_refpoints_xy: bool = False,
+                 num_patterns: int = 0,
+                 **kwargs) -> None:
+        self.iter_update = iter_update
+        self.random_refpoints_xy = random_refpoints_xy
+        self.num_patterns = num_patterns
+        if not isinstance(num_patterns, int):
```

Comment:
assert isinstance(num_patterns, int), f'num_patterns should be int but {num_parrerns}'.

mmdetection recommends using f-string.